doc/sphinx/language/gallina-extensions.rst

```diff
@@ -1696,50 +1643,69 @@ Declaring Implicit Arguments
-   inserted implicits are surrounded by curly braces.
-
-   After the above declaration is issued, implicit arguments can just
-   (and have to) be skipped in any expression involving an application
-   of :token:`qualid`.
-
-.. cmd:: Arguments @qualid : clear implicits
-   :name: Arguments (clear implicits)
-
-   This command clears implicit arguments.
-
-.. cmdv:: Global Arguments @qualid {* {| [ @name ] | { @name } | @name } }
-
-   This command is used to recompute the implicit arguments of
-   :token:`qualid` after ending of the current section if any, enforcing the
-   implicit arguments known from inside the section to be the ones
-   declared by the command.
-
-.. cmdv:: Local Arguments @qualid {* {| [ @name ] | { @name } | @name } }
-
-   When in a module, tell not to activate the
-   implicit arguments of :token:`qualid` declared by this command to contexts that
-   require the module.
-
-.. cmdv:: {? {| Global | Local } } Arguments @qualid {*, {+ {| [ @name ] | { @name } | @name } } }
-
-   For names of constants, inductive types,
-   constructors, lemmas which can only be applied to a fixed number of
-   arguments (this excludes for instance constants whose type is
-   polymorphic), multiple implicit arguments declarations can be given.
-   Depending on the number of arguments qualid is applied to in practice,
-   the longest applicable list of implicit arguments is used to select
-   which implicit arguments are inserted. For printing, the omitted
-   arguments are the ones of the longest list of implicit arguments of
-   the sequence.
-
-.. example::
+.. cmd:: Arguments @smart_qualid {* @argument_spec_block } {* , {* @more_implicits_block } } {? : {+, @arguments_modifier } }
+   :name: Arguments
+
+   .. insertprodn argument_spec_block arguments_modifier
+
+   .. prodn::
+      argument_spec_block ::= @argument_spec
+      | /
+      | &
+      | ( {+ @argument_spec } ) {? % @ident }
+      | [ {+ @argument_spec } ] {? % @ident }
+      | %{ {+ @argument_spec } %} {? % @ident }
+      argument_spec ::= {? ! } @name {? % @ident }
+      more_implicits_block ::= @name
+      | [ {+ @name } ]
+      | %{ {+ @name } %}
+      arguments_modifier ::= simpl nomatch
+      | simpl never
+      | default implicits
+      | clear bidirectionality hint
+      | clear implicits
+      | clear scopes
+      | clear scopes and implicits
+      | clear implicits and scopes
+      | rename
+      | assert
+      | extra scopes
+
+   This command sets implicit arguments *a posteriori*,
+   where the list of :n:`@name`\s is a prefix of the list of
+   arguments of :n:`@qualid` (in the :n:`@smart_qualid`).  Arguments in square
+   brackets are declared as implicit and arguments in curly brackets are declared as
+   maximally inserted.
+
+   After the command is issued, implicit arguments can and must be
+   omitted in any expression that applies :token:`qualid`.
+
+   This command supports the :attr:`local` and :attr:`global` attributes.
```

Comment:
FWIW AFAIK it was not explained in this file, but only as cmdvs in syntax-extensions.rst and under :attr:`local`.  I added the info here nonetheless, but still not exhaustive or all that clear.  Please take a look.

Reply:
No, it was explained here. Otherwise, I would not have left this comment:

```rst
 .. cmdv:: Global Arguments @qualid {* {| [ @name ] | { @name } | @name } }

    This command is used to recompute the implicit arguments of
    :token:`qualid` after ending of the current section if any, enforcing the
    implicit arguments known from inside the section to be the ones
    declared by the command.

.. cmdv:: Local Arguments @qualid {* {| [ @name ] | { @name } | @name } }

    When in a module, tell not to activate the
    implicit arguments of :token:`qualid` declared by this command to contexts that
    require the module.
```

That being said, this explanation was quite undecipherable and your addition is better IMHO.